### PR TITLE
Support publishing conformance test results to Azure DevOps

### PR DESCRIPTION
**Pull request by CodeMonkeyLeet**
# Description

- Update conformance test to output junit xml reports that can be consumed by Azure DevOps test analytics.
- Add conformance test workflow job to trigger pipeline in Azure DevOps that publishes the test results.
- Add Azure pipeline definition for publishing test results from a specified GitHub workflow run to Azure DevOps.

> ⚠ There are a number of maintainer actions necessary for this to be useful:
> - An ADO project to publish test results into.
>    - A new PAT from a Dapr service account in ADO (`AZURE_DEVOPS_TOKEN`) with _Read and Execute build_ permissions for GitHub to trigger the test result publishing workflow.
>    - A new build pipeline in the ADO project named `components-contrib.publish-conformance-results` based on the `.azuredevops/pipelines/publish-test-results.yml` definition from the dapr/components-contrib project after this change is merged.
> - A new PAT from a Dapr service account (`GITHUB_TOKEN`) with _public_repo_ permissions for ADO to access test logs

## Issue reference

POC using components-contrib conformance tests to address test dashboard requirement suggested in https://github.com/dapr/dapr/issues/3316

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by CodeMonkeyLeet** [comment]
@wcs1only @artursouza Putting this up so that any maintainer tasks can happen in parallel to code reviews.

File: .azuredevops/pipelines/publish-test-results.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  vmImage: ubuntu-latest
+
+steps:
+- script: |
+    python -m pip install --upgrade pip setuptools wheel
+    pip install ghapi
+  displayName: 'Install Python tools'
+
+- task: PythonScript@0
+  displayName: 'Download test artifacts'
```
Comment (jkotalik) [inline .azuredevops/pipelines/publish-test-results.yml:19]: Shot in the dark, but I'd be curious if you could just use the Download Pipeline/Build Artifacts AzDO task: https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/download-pipeline-artifact?view=azure-devops or https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/download-build-artifacts?view=azure-devops

The only reason I suggest that is that both github actions and azure devops artifacts use the same backend, I'd be curious if it just works. Though I don't know what the "pipeline" name would be.

Besides that, I can't think of anything else that would work better. Maybe reaching out to someone who works on Azure DevOps may be the next best step.

Comment (CodeMonkeyLeet) [reply .azuredevops/pipelines/publish-test-results.yml:19]: Thanks for the suggestion! I took a look at the [task definition](https://github.com/microsoft/azure-pipelines-tasks/blob/e6aab1600c8d50e44518ae0de54995f2716b66ad/Tasks/DownloadBuildArtifactsV0/task.json) for that and it's a wrapper for the TFS HTTP endpoint, so it doesn't look like it's possible to get through that to the underlying shared backend with GitHub. I'll see if I can find anyone on AzDevOps who might be able to help.

**Comment by CodeMonkeyLeet** [comment]
Per sync with maintainers, the desire is to switch to a pull-only model from the test dashboard service to remove the need for a GitHub action trigger. Closing this PR.

**Pull request closed without merging**